device/celestica/x86_64-cel_silverstone-r0/plugins/sfputil.py

```diff
@@ -379,9 +584,34 @@ def get_eeprom_dict(self, port_num):
                 sfp_data['interface'] = sfpi_obj.get_data_pretty()
 
                 # check if it is a 100G module
-                if sfp_data['interface']['data']['Identifier'] == 'QSFP28 or later':
+                if sfp_data['interface']['data']['Identifier'] not in [type_of_transceiver['18'], type_of_transceiver['19']]:
                     return self.get_qsfp_data(eeprom_ifraw)
 
+                # decode application advertisement
+                offset = 85
+                tbl = self.parse_media_type(eeprom_ifraw, offset)
+                ret = ""
+                if tbl is not None:
+                    app = 1
+                    hid = int(eeprom_ifraw[1 + offset], 16)
+                    while (hid != 0) and (hid != 0xff):
+                        (ht, mt) = self.parse_application(tbl, eeprom_ifraw[1 + offset], eeprom_ifraw[2 + offset])
+                        ret += "\n            {0}: {1} | {2}".format(app, ht, mt)
+                        app += 1
+                        offset += 4
+                        hid = int(eeprom_ifraw[1 + offset], 16)
+                if len(ret) > 0:
+                    sfp_data['interface']['data']['Application Advertisement'] = ret
```

Comment:
ret format is a bit strange?

Reply:
It's an one-line message for the application modes, in the current implementation, it includes host-interface-id and media-interface-id, and it's separated by '|'
e.g.
Ethernet96: SFP EEPROM detected
        Application Advertisement:
            1: 400G CR8 | Copper cable
            2: 200GBASE-CR4 (Clause 136) | Copper cable
            3: 100GBASE-CR4 (Clause 92) | Copper cable
            4: 50GBASE-CR (Clause 126) | Copper cable
            5: IB HDR (Arch.Spec.Vol.2) | Copper cable
            6: 25GBASE-CR CA-L (Clause 110) | Copper cable
            7: IB EDR (Arch.Spec.Vol.2) | Copper cable